### PR TITLE
annotation: don't change slide/sheet while editing comment 

### PR DESCRIPTION
**Pull request by lpranam**
* Target version: master 

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: browser/src/control/Control.PartsPreview.js [hunk 1]
```diff
@@ -499,6 +499,11 @@ L.Control.PartsPreview = L.Control.extend({
 	},
 
 	_setPart: function (e) {
+		if (cool.Comment.isAnyEdit()) {
```
Comment (eszkadev) [inline browser/src/control/Control.PartsPreview.js:502]: All of these different actions from different components are using `this._map.setPart` in the end. Could it be possible to fix it in one place there?

Or these actions have other side effects that could make it more complicated?

File: browser/src/control/Control.Tabs.js [hunk 2]
```diff
@@ -271,6 +271,11 @@ L.Control.Tabs = L.Control.extend({
 	},
 
 	_setPart: function (e) {
+		if (cool.Comment.isAnyEdit()) {
```
Comment (eszkadev) [inline browser/src/control/Control.Tabs.js:274]: same here

**Review by eszkadev — COMMENTED** [review]
what about Calc? did you test if the bug happens also there?

**Comment by eszkadev** [comment]
btw: In Calc when I click outside "spreadsheet area", eg. "Bold" button in sidebar - comment is not autosaved

Maybe we should also block sending unoCommands when comment is unser editing? and autosave too.
example what we do for dialogs when they are already opened: https://github.com/CollaboraOnline/online/blob/master/browser/src/control/Toolbar.js#L402

File: browser/src/control/Control.PartsPreview.js (hunk 1 above)
Comment (lpranam) [reply browser/src/control/Control.PartsPreview.js:502]: It makes it more complicated as these methods are calling other functions like selectPart and deselectAll or _scrollViewToPartPosition. I am afraid I might go into a rabbit hole if we don't stop these actions at their starting point.

Comment (eszkadev) [reply browser/src/control/Control.PartsPreview.js:502]: I see

**Comment by lpranam** [comment]
> btw: In Calc when I click outside "spreadsheet area", eg. "Bold" button in sidebar - comment is not autosaved
> 
> Maybe we should also block sending unoCommands when comment is unser editing? and autosave too. example what we do for dialogs when they are already opened: https://github.com/CollaboraOnline/online/blob/master/browser/src/control/Toolbar.js#L402

This is an interesting case... comment never lost focus so never autosave. But on the other hand I don't wish to block the uno command as it defeats the purpose of the autosave which is to allow editing or doing anything else...

**Comment by lpranam** [comment]
hmmm if it doesn't even lose focus why to force it to autosave, I was testing and I am tempted to keep it as it is.. what's your opinion on that @eszkadev ?

**Comment by eszkadev** [comment]
I don't know, just pointing what I noticed. I think it's not directly related to this fix so no need to solve it here.

But please fix `ctrl + alt + page UP/DOWN` case

**Review by eszkadev — APPROVED** [review]
let's get this in and prepare followup for shortcuts later

**Comment by lpranam** [comment]
I had added check in setPart so even shortcut would be blocked :)

**Pull request merged**